src/js/components/ToggleGroup/ToggleGroup.js

```diff
@@ -0,0 +1,190 @@
+      // assume that value might not be ordered the same as options
+      defaultIndex = values.indexOf(
+        values.find((option) => value.includes(option)),
+      );
+    }
+    return defaultIndex;
+  }, [value, values]);
+
+  const [focusableIndex, setFocusableIndex] = useState(() =>
+    getFocusableIndex(),
+  );
+
+  const onNext = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex + 1 <= options.length - 1 ? focusableIndex + 1 : 0;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const onPrevious = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex - 1 >= 0 ? focusableIndex - 1 : options.length - 1;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const handleToggle = (event, option) => {
+    const adjustedEvent = event;
+    let nextValue;
+    if (!multiple) {
+      nextValue = option;
+    } else {
+      nextValue = value.includes(option)
+        ? value.filter((item) => item !== option)
+        : [...value, option];
+    }
+    adjustedEvent.value = nextValue;
+    setValue(adjustedEvent);
+  };
+
+  return (
+    <Keyboard
+      onUp={onPrevious}
+      onDown={onNext}
+      onLeft={onPrevious}
+      onRight={onNext}
+    >
+      <Box
+        ref={ref}
+        direction="row"
+        alignSelf="start"
+        role={multiple ? 'group' : 'radiogroup'}
+        onBlur={(e) => {
+          if (!ref?.current.contains(e.relatedTarget)) {
+            setFocusableIndex(getFocusableIndex());
+          }
+        }}
+        {...theme.toggleGroup.container}
+        // match button rounding
+        responsive={false}
+        {...rest}
+      >
+        {options.map((option, index) => {
+          let label;
+          let icon;
+          let optionValue;
+          if (typeof option === 'object') {
+            icon = option.icon;
+            label = option.label;
+            optionValue = option.value;
+          } else {
+            label = option;
+            optionValue = option;
+          }
+          const active = Array.isArray(value)
+            ? !!value.includes(optionValue)
+            : value === optionValue;
+          let round = 0;
+          // round corners of first and last buttons to match container
+          if (
+            typeof theme.toggleGroup.container.round === 'string' &&
+            (index === 0 || index === options.length - 1)
+          ) {
+            round = {
+              corner: index === 0 ? 'left' : 'right',
+              size: theme.toggleGroup.container.round,
+            };
+          }
+
+          const iconOnly = icon && !label;
+          let pad;
+          if (iconOnly && theme.toggleGroup.button?.iconOnly?.pad) {
+            pad = theme.toggleGroup.button.iconOnly.pad;
+          } else pad = theme.toggleGroup.button.pad;
+
+          return (
+            <Box
```

Comment:
Do we need to think about some sort of flex={false}?

![Screenshot 2024-04-17 at 3 35 34 PM](https://github.com/grommet/grommet/assets/1756948/3d342f93-a9de-456e-9eb9-7061bd281445)

Reply:
Added `flex={false}`.